### ISSUE TITLE
[Windows] Text field accessibility is broken

### ISSUE THREAD

**Issue by chrisnorman7** [flutter/flutter#103464]
Text fields are basically unusable, and it is often to write text in some other program, then paste it into Flutter apps.

I could have sworn there was already an issue for this, but I am unable to find it, so sorry if this is a duplicate.

## Steps to Reproduce

1. Create an app with the below code.
2. Run `flutter run -d windows`.
3. With a screen reader (such as [NVDA](https://www.nvaccess.org/download)), enter some text, navigate through the text with the arrow keys, try and find an arbitrary point to add more text, ETC.
## Problems

- Moving through the text field with the arrow keys (or home and end) speaks nothing (#95556).
- When typing new characters, the text of the field is spoken in its entirety.
- When deleting text with the backspace or delete keys, the text is spoken in its entirety.
- When selecting text with shift and the arrow keys, nothing is spoken.

## Logs

<details>
<summary>Logs</summary>

```
[  +44 ms] executing: [C:\Users\chris\flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[ +193 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] a259fcb64e18f96b55d074ef17cb0acec5e47bb3
[        ] executing: [C:\Users\chris\flutter/] git tag --points-at a259fcb64e18f96b55d074ef17cb0acec5e47bb3
[  +51 ms] Exit code 0 from: git tag --points-at a259fcb64e18f96b55d074ef17cb0acec5e47bb3
[        ] executing: [C:\Users\chris\flutter/] git describe --match *.*.* --long --tags a259fcb64e18f96b55d074ef17cb0acec5e47bb3
[  +76 ms] Exit code 0 from: git describe --match *.*.* --long --tags a259fcb64e18f96b55d074ef17cb0acec5e47bb3
[        ] 2.12.0-4.0.pre-1023-ga259fcb64e
[  +32 ms] executing: [C:\Users\chris\flutter/] git rev-parse --abbrev-ref --symbolic @{upstream}
[  +34 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{upstream}
[        ] origin/master
[        ] executing: [C:\Users\chris\flutter/] git ls-remote --get-url origin
[  +33 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +62 ms] executing: [C:\Users\chris\flutter/] git rev-parse --abbrev-ref HEAD
[  +34 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] master
[  +59 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +2 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +41 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +2 ms] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +55 ms] Skipping pub get: version match.
[  +73 ms] Generating C:\Users\chris\src\worldsmith_examples\custom_keys\poc\android\app\src\main\java\io\flutter\plugins\GeneratedPluginRegistrant.java
[  +69 ms] Initializing file store
[   +7 ms] Skipping target: gen_localizations
[   +4 ms] gen_dart_plugin_registrant: Starting due to {InvalidatedReasonKind.inputChanged: The following inputs have updated contents: C:\Users\chris\src\worldsmith_examples\custom_keys\poc\.dart_tool\package_config_subset}
[  +15 ms] gen_dart_plugin_registrant: Complete
[   +1 ms] Skipping target: _composite
[   +1 ms] complete
[   +4 ms] Launching lib\main.dart on Windows in debug mode...
[   +3 ms] C:\Users\chris\flutter\bin\cache\dart-sdk\bin\dart.exe --disable-dart-dev C:\Users\chris\flutter\bin\cache\dart-sdk\bin\snapshots\frontend_server.dart.snapshot --sdk-root C:\Users\chris\flutter\bin\cache\artifacts\engine\common\flutter_patched_sdk/ --incremental --target=flutter --debugger-module-names --experimental-emit-debug-metadata -DFLUTTER_WEB_AUTO_DETECT=true --output-dill C:\Users\chris\AppData\Local\Temp\flutter_tools.398dea73\flutter_tool.621d315d\app.dill --packages C:\Users\chris\src\worldsmith_examples\custom_keys\poc\.dart_tool\package_config.json -Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation --filesystem-scheme org-dartlang-root --initialize-from-dill build\c075001b96339384a97db4862b8ab8db.cache.dill.track.dill --flutter-widget-cache --enable-experiment=alternative-invalidation-strategy
[  +17 ms] executing: C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe -format json -products * -utf8 -latest -version 16 -requires Microsoft.VisualStudio.Workload.NativeDesktop Microsoft.VisualStudio.Component.VC.Tools.x86.x64 Microsoft.VisualStudio.Component.VC.CMake.Project
[  +32 ms] Exit code 0 from: C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe -format json -products * -utf8 -latest -version 16 -requires Microsoft.VisualStudio.Workload.NativeDesktop Microsoft.VisualStudio.Component.VC.Tools.x86.x64 Microsoft.VisualStudio.Component.VC.CMake.Project
[        ] [
             {
               "instanceId": "48f70a3c",
               "installDate": "2022-02-27T12:24:06Z",
               "installationName": "VisualStudio/17.1.0+32210.238",
               "installationPath": "C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise",
               "installationVersion": "17.1.32210.238",
               "productId": "Microsoft.VisualStudio.Product.Enterprise",
               "productPath": "C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\Common7\\IDE\\devenv.exe",
               "state": 4294967295,
               "isComplete": true,
               "isLaunchable": true,
               "isPrerelease": false,
               "isRebootRequired": false,
               "displayName": "Visual Studio Enterprise 2022",
               "description": "Scalable, end-to-end solution for teams of any size",
               "channelId": "VisualStudio.17.Release",
               "channelUri": "https://aka.ms/vs/17/release/channel",
               "enginePath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\resources\\app\\ServiceHub\\Services\\Microsoft.VisualStudio.Setup.Service",
               "installedChannelUri": "https://aka.ms/vs/17/release/channel",
               "releaseNotes": "https://docs.microsoft.com/en-us/visualstudio/releases/2022/release-notes-v17.1#17.1.0",
               "thirdPartyNotices": "https://go.microsoft.com/fwlink/?LinkId=661288",
               "updateDate": "2022-02-27T12:24:06.5628333Z",
               "catalog": {
                 "buildBranch": "d17.1",
                 "buildVersion": "17.1.32210.238",
                 "id": "VisualStudio/17.1.0+32210.238",
                 "localBuild": "build-lab",
                 "manifestName": "VisualStudio",
                 "manifestType": "installer",
                 "productDisplayVersion": "17.1.0",
                 "productLine": "Dev17",
                 "productLineVersion": "2022",
                 "productMilestone": "RTW",
                 "productMilestoneIsPreRelease": "False",
                 "productName": "Visual Studio",
                 "productPatchVersion": "0",
                 "productPreReleaseMilestoneSuffix": "7.0",
                 "productSemanticVersion": "17.1.0+32210.238",
                 "requiredEngineVersion": "3.1.2196.8931"
               },
               "properties": {
                 "campaignId": "2030",
                 "channelManifestId": "VisualStudio.17.Release/17.1.0+32210.238",
                 "nickname": "2",
                 "setupEngineFilePath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\setup.exe"
               }
             }
           ]
[   +1 ms] Building Windows application...
[  +13 ms] <- compile package:poc/main.dart
[   +1 ms] executing: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe -S C:\Users\chris\src\worldsmith_examples\custom_keys\poc\windows -B build\windows -G Visual Studio 17 2022
[  +48 ms] -- Configuring done
[ +151 ms] -- Generating done
[  +16 ms] -- Build files have been written to: C:/Users/chris/src/worldsmith_examples/custom_keys/poc/build/windows
[  +21 ms] executing: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe --build build\windows --config Debug --target INSTALL --verbose
[  +91 ms] Microsoft (R) Build Engine version 17.1.0+ae57d105c for .NET Framework
[        ] Copyright (C) Microsoft Corporation. All rights reserved.
[  +48 ms] Build started 11/05/2022 06:16:12.
[ +148 ms] Project "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\INSTALL.vcxproj" on node 1 (default targets).
[        ] Project "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\INSTALL.vcxproj" (1) is building "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\ZERO_CHECK.vcxproj" (2) on node 1 (default targets).
[        ] InitializeBuildStatus:
[        ]   Creating "x64\Debug\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[  +15 ms] CustomBuild:
[        ]   All outputs are up-to-date.
[  +37 ms]   Checking Build System
[  +54 ms] FinalizeBuildStatus:
[        ]   Deleting file "x64\Debug\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild".
[        ]   Touching "x64\Debug\ZERO_CHECK\ZERO_CHECK.tlog\ZERO_CHECK.lastbuildstate".
[        ] Done Building Project "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\ZERO_CHECK.vcxproj" (default targets).
[  +60 ms] Project "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\INSTALL.vcxproj" (1) is building "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\ALL_BUILD.vcxproj" (3) on node 1 (default targets).
[        ] Project "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\ALL_BUILD.vcxproj" (3) is building "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\flutter\flutter_wrapper_app.vcxproj" (4) on node 1 (default targets).
[        ] Project "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\flutter\flutter_wrapper_app.vcxproj" (4) is building "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\flutter\flutter_assemble.vcxproj" (5) on node 1 (default targets).
[        ] InitializeBuildStatus:
[        ]   Creating "x64\Debug\flutter_assemble\flutter_assemble.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[   +4 ms] CustomBuild:
[        ]   "The build of 'C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\CMakeFiles\5b19e00c10f0cab27b3f2e7c5d8e8408\flutter_assemble.rule' depends on 'C:\USERS\CHRIS\SRC\WORLDSMITH_EXAMPLES\CUSTOM_KEYS\POC\WINDOWS\FLUTTER\EPHEMERAL\FLUTTER_WINDOWS.DLL' which is produced by the build of 'C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\CMakeFiles\774986b364a318e3a443415865be5f9f\flutter_windows.dll.rule'. The items cannot be built in parallel."
[  +41 ms]   Generating C:/Users/chris/src/worldsmith_examples/custom_keys/poc/windows/flutter/ephemeral/flutter_windows.dll, C:/Users/chris/src/worldsmith_examples/custom_keys/poc/windows/flutter/ephemeral/flutter_export.h, C:/Users/chris/src/worldsmith_examples/custom_keys/poc/windows/flutter/ephemeral/flutter_windows.h, C:/Users/chris/src/worldsmith_examples/custom_keys/poc/windows/flutter/ephemeral/flutter_messenger.h, C:/Users/chris/src/worldsmith_examples/custom_keys/poc/windows/flutter/ephemeral/flutter_plugin_registrar.h, C:/Users/chris/src/worldsmith_examples/custom_keys/poc/windows/flutter/ephemeral/flutter_texture_registrar.h, C:/Users/chris/src/worldsmith_examples/custom_keys/poc/windows/flutter/ephemeral/cpp_client_wrapper/core_implementations.cc, C:/Users/chris/src/worldsmith_examples/custom_keys/poc/windows/flutter/ephemeral/cpp_client_wrapper/standard_codec.cc, C:/Users/chris/src/worldsmith_examples/custom_keys/poc/windows/flutter/ephemeral/cpp_client_wrapper/plugin_registrar.cc, C:/Users/chris/src/worldsmith_examples/custom_keys/poc/windows/flutter/ephemeral/cpp_client_wrapper/flutter_engine.cc, C:/Users/chris/src/worldsmith_examples/custom_keys/poc/windows/flutter/ephemeral/cpp_client_wrapper/flutter_view_controller.cc, _phony_
[ +693 ms]   [  +44 ms] executing: [C:\Users\chris\flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[ +187 ms]   [ +201 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ]   [        ] a259fcb64e18f96b55d074ef17cb0acec5e47bb3
[        ]   [        ] executing: [C:\Users\chris\flutter/] git tag --points-at a259fcb64e18f96b55d074ef17cb0acec5e47bb3
[  +58 ms]   [  +57 ms] Exit code 0 from: git tag --points-at a259fcb64e18f96b55d074ef17cb0acec5e47bb3
[        ]   [   +1 ms] executing: [C:\Users\chris\flutter/] git describe --match *.*.* --long --tags a259fcb64e18f96b55d074ef17cb0acec5e47bb3
[  +85 ms]   [  +84 ms] Exit code 0 from: git describe --match *.*.* --long --tags a259fcb64e18f96b55d074ef17cb0acec5e47bb3
[        ]   [        ] 2.12.0-4.0.pre-1023-ga259fcb64e
[  +42 ms]   [  +42 ms] executing: [C:\Users\chris\flutter/] git rev-parse --abbrev-ref HEAD
[  +46 ms]   [  +45 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ]   [        ] master
[  +47 ms]   [  +48 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +1 ms]   [   +2 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +64 ms]   [  +64 ms] executing: [C:\Users\chris\flutter/] git rev-parse --abbrev-ref --symbolic @{upstream}
[  +41 ms]   [  +41 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{upstream}
[        ]   [        ] origin/master
[        ]   [        ] executing: [C:\Users\chris\flutter/] git ls-remote --get-url origin
[  +36 ms]   [  +36 ms] Exit code 0 from: git ls-remote --get-url origin
[        ]   [        ] https://github.com/flutter/flutter.git
[        ]   [        ] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[   +1 ms]   [   +1 ms] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[  +24 ms]   [  +24 ms] Initializing file store
[   +8 ms]   [   +8 ms] Done initializing file store
[  +26 ms]   [  +26 ms] Skipping target: gen_localizations
[   +6 ms]   [   +6 ms] Skipping target: gen_dart_plugin_registrant
[+1655 ms]   [+1654 ms] Skipping target: unpack_windows
[ +312 ms]   [ +312 ms] Skipping target: kernel_snapshot
[ +162 ms]   [ +163 ms] Skipping target: debug_bundle_windows_assets
[        ]   [        ] Persisting file store
[   +6 ms]   [   +6 ms] Done persisting file store
[   +3 ms]   [   +3 ms] build succeeded.
[   +5 ms]   [   +5 ms] "flutter assemble" took 2,379ms.
[  +10 ms]   [  +11 ms] ensureAnalyticsSent: 8ms
[        ]   [        ] Running shutdown hooks
[        ]   [        ] Shutdown hooks complete
[        ]   [        ] exiting with code 0
[  +41 ms] FinalizeBuildStatus:
[        ]   Deleting file "x64\Debug\flutter_assemble\flutter_assemble.tlog\unsuccessfulbuild".
[        ]   Touching "x64\Debug\flutter_assemble\flutter_assemble.tlog\flutter_assemble.lastbuildstate".
[        ] Done Building Project "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\flutter\flutter_assemble.vcxproj" (default targets).
[   +4 ms] InitializeBuildStatus:
[        ]   Creating "flutter_wrapper_app.dir\Debug\flutter_.F599F3CB.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[   +1 ms] CustomBuild:
[        ]   All outputs are up-to-date.
[  +37 ms] ClCompile:
[        ]   All outputs are up-to-date.
[  +13 ms] Lib:
[        ]   All outputs are up-to-date.
[        ]   flutter_wrapper_app.vcxproj -> C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\flutter\Debug\flutter_wrapper_app.lib
[   +6 ms] FinalizeBuildStatus:
[        ]   Deleting file "flutter_wrapper_app.dir\Debug\flutter_.F599F3CB.tlog\unsuccessfulbuild".
[        ]   Touching "flutter_wrapper_app.dir\Debug\flutter_.F599F3CB.tlog\flutter_wrapper_app.lastbuildstate".
[        ] Done Building Project "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\flutter\flutter_wrapper_app.vcxproj" (default targets).
[  +18 ms] Project "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\ALL_BUILD.vcxproj" (3) is building "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\flutter\flutter_wrapper_plugin.vcxproj" (6) on node 1 (default targets).
[        ] InitializeBuildStatus:
[        ]   Creating "flutter_wrapper_plugin.dir\Debug\flutter_.0416986A.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[   +1 ms] CustomBuild:
[        ]   All outputs are up-to-date.
[   +5 ms] ClCompile:
[        ]   All outputs are up-to-date.
[   +2 ms] Lib:
[        ]   All outputs are up-to-date.
[        ]   flutter_wrapper_plugin.vcxproj -> C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\flutter\Debug\flutter_wrapper_plugin.lib
[   +5 ms] FinalizeBuildStatus:
[        ]   Deleting file "flutter_wrapper_plugin.dir\Debug\flutter_.0416986A.tlog\unsuccessfulbuild".
[        ]   Touching "flutter_wrapper_plugin.dir\Debug\flutter_.0416986A.tlog\flutter_wrapper_plugin.lastbuildstate".
[        ] Done Building Project "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\flutter\flutter_wrapper_plugin.vcxproj" (default targets).
[  +16 ms] Project "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\ALL_BUILD.vcxproj" (3) is building "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\runner\poc.vcxproj" (7) on node 1 (default targets).
[        ] InitializeBuildStatus:
[        ]   Creating "poc.dir\Debug\poc.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[   +1 ms] CustomBuild:
[        ]   All outputs are up-to-date.
[  +13 ms] ClCompile:
[        ]   C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.31.31103\bin\HostX64\x64\CL.exe /c /IC:\Users\chris\src\worldsmith_examples\custom_keys\poc\windows /IC:\Users\chris\src\worldsmith_examples\custom_keys\poc\windows\flutter\ephemeral /IC:\Users\chris\src\worldsmith_examples\custom_keys\poc\windows\flutter\ephemeral\cpp_client_wrapper\include /Zi /nologo /W4 /WX /diagnostics:column /Od /Ob0 /D _UNICODE /D UNICODE /D WIN32 /D _WINDOWS /D _HAS_EXCEPTIONS=0 /D _DEBUG /D NOMINMAX /D UNICODE /D _UNICODE /D "CMAKE_INTDIR=\"Debug\"" /Gm- /EHsc /RTC1 /MDd /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /GR /std:c++17 /Fo"poc.dir\Debug\\" /Fd"poc.dir\Debug\vc143.pdb" /external:W4 /Gd /TP /wd4100 /errorReport:queue C:\Users\chris\src\worldsmith_examples\custom_keys\poc\windows\flutter\generated_plugin_registrant.cc C:\Users\chris\src\worldsmith_examples\custom_keys\poc\windows\runner\flutter_window.cpp
[  +26 ms]   generated_plugin_registrant.cc
[ +248 ms]   flutter_window.cpp
[ +464 ms]   Generating Code...
[  +48 ms] ResourceCompile:
[        ]   All outputs are up-to-date.
[  +20 ms] Link:
[        ]   C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.31.31103\bin\HostX64\x64\link.exe /ERRORREPORT:QUEUE /OUT:"C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\runner\Debug\poc.exe" /INCREMENTAL /ILK:"poc.dir\Debug\poc.ilk" /NOLOGO ..\flutter\Debug\flutter_wrapper_app.lib C:\Users\chris\src\worldsmith_examples\custom_keys\poc\windows\flutter\ephemeral\flutter_windows.dll.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed /manifestinput:C:\Users\chris\src\worldsmith_examples\custom_keys\poc\windows\runner\runner.exe.manifest /DEBUG /PDB:"C:/Users/chris/src/worldsmith_examples/custom_keys/poc/build/windows/runner/Debug/poc.pdb" /SUBSYSTEM:WINDOWS /TLBID:1 /DYNAMICBASE /NXCOMPAT /IMPLIB:"C:/Users/chris/src/worldsmith_examples/custom_keys/poc/build/windows/runner/Debug/poc.lib" /MACHINE:X64  /machine:x64 poc.dir\Debug\Runner.res
[        ]   poc.dir\Debug\flutter_window.obj
[        ]   poc.dir\Debug\main.obj
[        ]   poc.dir\Debug\utils.obj
[        ]   poc.dir\Debug\win32_window.obj
[        ]   poc.dir\Debug\generated_plugin_registrant.obj
[ +108 ms]   poc.vcxproj -> C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\runner\Debug\poc.exe
[   +6 ms] FinalizeBuildStatus:
[        ]   Deleting file "poc.dir\Debug\poc.tlog\unsuccessfulbuild".
[        ]   Touching "poc.dir\Debug\poc.tlog\poc.lastbuildstate".
[        ] Done Building Project "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\runner\poc.vcxproj" (default targets).
[   +5 ms] InitializeBuildStatus:
[        ]   Creating "x64\Debug\ALL_BUILD\ALL_BUILD.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[   +1 ms] CustomBuild:
[        ]   All outputs are up-to-date.
[  +19 ms]   Building Custom Rule C:/Users/chris/src/worldsmith_examples/custom_keys/poc/windows/CMakeLists.txt
[  +24 ms] FinalizeBuildStatus:
[        ]   Deleting file "x64\Debug\ALL_BUILD\ALL_BUILD.tlog\unsuccessfulbuild".
[        ]   Touching "x64\Debug\ALL_BUILD\ALL_BUILD.tlog\ALL_BUILD.lastbuildstate".
[        ] Done Building Project "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\ALL_BUILD.vcxproj" (default targets).
[   +3 ms] InitializeBuildStatus:
[        ]   Creating "x64\Debug\INSTALL\INSTALL.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[  +25 ms] PostBuildEvent:
[        ]   setlocal
[        ]   "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -DBUILD_TYPE=Debug -P cmake_install.cmake
[        ]   if %errorlevel% neq 0 goto :cmEnd
[        ]   :cmEnd
[        ]   endlocal & call :cmErrorLevel %errorlevel% & goto :cmDone
[        ]   :cmErrorLevel
[        ]   exit /b %1
[        ]   :cmDone
[        ]   if %errorlevel% neq 0 goto :VCEnd
[        ]   :VCEnd
[  +25 ms]   -- Install configuration: "Debug"
[  +13 ms]   -- Up-to-date: C:/Users/chris/src/worldsmith_examples/custom_keys/poc/build/windows/runner/Debug/data/icudtl.dat
[        ]   -- Up-to-date: C:/Users/chris/src/worldsmith_examples/custom_keys/poc/build/windows/runner/Debug/flutter_windows.dll
[   +7 ms]   -- Installing: C:/Users/chris/src/worldsmith_examples/custom_keys/poc/build/windows/runner/Debug/data/flutter_assets
[        ]   -- Installing: C:/Users/chris/src/worldsmith_examples/custom_keys/poc/build/windows/runner/Debug/data/flutter_assets/AssetManifest.json
[   +1 ms]   -- Installing: C:/Users/chris/src/worldsmith_examples/custom_keys/poc/build/windows/runner/Debug/data/flutter_assets/FontManifest.json
[        ]   -- Installing: C:/Users/chris/src/worldsmith_examples/custom_keys/poc/build/windows/runner/Debug/data/flutter_assets/fonts
[        ]   -- Installing: C:/Users/chris/src/worldsmith_examples/custom_keys/poc/build/windows/runner/Debug/data/flutter_assets/fonts/MaterialIcons-Regular.otf
[   +5 ms]   -- Installing: C:/Users/chris/src/worldsmith_examples/custom_keys/poc/build/windows/runner/Debug/data/flutter_assets/kernel_blob.bin
[  +94 ms]   -- Installing: C:/Users/chris/src/worldsmith_examples/custom_keys/poc/build/windows/runner/Debug/data/flutter_assets/NOTICES.Z
[   +1 ms]   -- Installing: C:/Users/chris/src/worldsmith_examples/custom_keys/poc/build/windows/runner/Debug/data/flutter_assets/packages
[        ]   -- Installing: C:/Users/chris/src/worldsmith_examples/custom_keys/poc/build/windows/runner/Debug/data/flutter_assets/packages/cupertino_icons
[        ]   -- Installing: C:/Users/chris/src/worldsmith_examples/custom_keys/poc/build/windows/runner/Debug/data/flutter_assets/packages/cupertino_icons/assets
[        ]   -- Installing: C:/Users/chris/src/worldsmith_examples/custom_keys/poc/build/windows/runner/Debug/data/flutter_assets/packages/cupertino_icons/assets/CupertinoIcons.ttf
[   +1 ms]   -- Installing: C:/Users/chris/src/worldsmith_examples/custom_keys/poc/build/windows/runner/Debug/data/flutter_assets/shaders
[        ]   -- Installing: C:/Users/chris/src/worldsmith_examples/custom_keys/poc/build/windows/runner/Debug/data/flutter_assets/shaders/ink_sparkle.frag
[  +13 ms] FinalizeBuildStatus:
[        ]   Deleting file "x64\Debug\INSTALL\INSTALL.tlog\unsuccessfulbuild".
[        ]   Touching "x64\Debug\INSTALL\INSTALL.tlog\INSTALL.lastbuildstate".
[        ] Done Building Project "C:\Users\chris\src\worldsmith_examples\custom_keys\poc\build\windows\INSTALL.vcxproj" (default targets).
[   +1 ms] Build succeeded.
[        ]     0 Warning(s)
[        ]     0 Error(s)
[        ] Time Elapsed 00:00:05.25
[  +15 ms] Building Windows application... (completed in 5.7s)
[ +196 ms] Observatory URL on device: http://127.0.0.1:29052/Wx0dO1Hii2o=/
[   +3 ms] Caching compiled dill
[  +20 ms] Connecting to service protocol: http://127.0.0.1:29052/Wx0dO1Hii2o=/
[  +90 ms] Launching a Dart Developer Service (DDS) instance at http://127.0.0.1:0, connecting to VM service at http://127.0.0.1:29052/Wx0dO1Hii2o=/.
[  +43 ms] DDS is listening at http://127.0.0.1:29055/Gxv3T__16AI=/.
[  +31 ms] Successfully connected to service protocol: http://127.0.0.1:29052/Wx0dO1Hii2o=/
[  +13 ms] DevFS: Creating new filesystem on the device (null)
[  +10 ms] DevFS: Created new filesystem on the device (file:///C:/Users/chris/AppData/Local/Temp/pocee1c7eed/poc/)
[   +1 ms] Updating assets
[  +50 ms] Syncing files to device Windows...
[        ] <- reset
[        ] Compiling dart to kernel with 0 updated files
[        ] Processing bundle.
[        ] <- recompile package:poc/main.dart 73e0cbc4-8988-4a7d-93f3-a3b6822b93c8
[        ] <- 73e0cbc4-8988-4a7d-93f3-a3b6822b93c8
[   +1 ms] Bundle processing done.
[  +44 ms] Updating files.
[        ] DevFS: Sync finished
[        ] Syncing files to device Windows... (completed in 48ms)
[        ] Synced 0.0MB.
[        ] <- accept
[   +1 ms] Connected to _flutterView/0x2ab2cb3d020.
[   +1 ms] Flutter run key commands.
[        ] r Hot reload. 
[        ] R Hot restart.
[        ] h List all available interactive commands.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ]  Running with sound null safety 
[        ] An Observatory debugger and profiler on Windows is available at: http://127.0.0.1:29055/Gxv3T__16AI=/
[ +168 ms] The Flutter DevTools debugger and profiler on Windows is available at: http://127.0.0.1:9102?uri=http://127.0.0.1:29055/Gxv3T__16AI=/
[+9004 ms] DevFS: Deleting filesystem on the device (file:///C:/Users/chris/AppData/Local/Temp/pocee1c7eed/poc/)
[   +3 ms] DevFS: Deleted filesystem on the device (file:///C:/Users/chris/AppData/Local/Temp/pocee1c7eed/poc/)
[   +5 ms] Application finished.
[   +3 ms] "flutter run" took 15,716ms.
[  +10 ms] Service protocol connection closed.
[   +1 ms] ensureAnalyticsSent: 10ms
[        ] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 0
```

```
[ÔêÜ] Flutter (Channel master, 2.13.0-0.0.pre.1023, on Microsoft Windows [Version 10.0.22000.613], locale en-GB)
    ÔÇó Flutter version 2.13.0-0.0.pre.1023 at C:\Users\chris\flutter
    ÔÇó Upstream repository https://github.com/flutter/flutter.git
    ÔÇó Framework revision a259fcb64e (63 minutes ago), 2022-05-11 00:14:06 -0400
    ÔÇó Engine revision 6291508027
    ÔÇó Dart version 2.18.0 (build 2.18.0-103.0.dev)
    ÔÇó DevTools version 2.13.1

[ÔêÜ] Android toolchain - develop for Android devices (Android SDK version 31.0.0-rc2)
    ÔÇó Android SDK at C:\Users\chris\AppData\Local\Android\sdk
    ÔÇó Platform android-31, build-tools 31.0.0-rc2
    ÔÇó Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    ÔÇó Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    ÔÇó All Android licenses accepted.

[ÔêÜ] Chrome - develop for the web
    ÔÇó Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[ÔêÜ] Visual Studio - develop for Windows (Visual Studio Enterprise 2022 17.1.0)
    ÔÇó Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Enterprise
    ÔÇó Visual Studio Enterprise 2022 version 17.1.32210.238
    ÔÇó Windows 10 SDK version 10.0.22000.0

[ÔêÜ] Android Studio (version 2020.3)
    ÔÇó Android Studio at C:\Program Files\Android\Android Studio
    ÔÇó Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    ÔÇó Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    ÔÇó Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[ÔêÜ] VS Code, 64-bit edition (version 1.67.0)
    ÔÇó VS Code at C:\Program Files\Microsoft VS Code
    ÔÇó Flutter extension version 3.27.0

[ÔêÜ] Connected device (3 available)
    ÔÇó Windows (desktop) ÔÇó windows ÔÇó windows-x64    ÔÇó Microsoft Windows [Version 10.0.22000.613]
    ÔÇó Chrome (web)      ÔÇó chrome  ÔÇó web-javascript ÔÇó Google Chrome 101.0.4951.54
    ÔÇó Edge (web)        ÔÇó edge    ÔÇó web-javascript ÔÇó Microsoft Edge 100.0.1185.44

[ÔêÜ] HTTP Host Availability
    ÔÇó All required HTTP hosts are available

ÔÇó No issues found!
```

</details>

## Code

```
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  late final TextEditingController _controller;

  /// Initialise the controller.
  @override
  void initState() {
    super.initState();
    _controller = TextEditingController(
        text:
            'Hello,\nThis is Flutter.\n\nThis text field is incredibly hard to interact with while using a screen reader.');
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text(widget.title),
      ),
      body: Center(
        // Center is a layout widget. It takes a single child and positions it
        // in the middle of the parent.
        child: TextField(
          autofocus: true,
          controller: _controller,
        ),
      ),
    );
  }

  /// Dispose of the widget.
  @override
  void dispose() {
    super.dispose();
    _controller.dispose();
  }
}
```

**Comment by darshankawar** [comment]
Thanks for the detailed report. From the list of problems mentioned, apart from first point for which there's an already open issue and linked above, I was able to replicate below scenarios:

```
When typing new characters, the text of the field is spoken in its entirety.
When deleting text with the backspace or delete keys, the text is spoken in its entirety.
When selecting text with shift and the arrow keys, nothing is spoken.
```

<details>
<summary> stable, master flutter doctor </summary>

```

[√] Flutter (Channel stable, 2.10.5, on Microsoft Windows [Version 10.0.19042.1526], locale en-US)
[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.2)
[√] Android Studio (version 2021.1)
[√] VS Code (version 1.62.3)
[√] Connected device (3 available)
[√] HTTP Host Availability

• No issues found!

[√] Flutter (Channel master, 2.13.0-0.0.pre.1023, on Microsoft Windows [Version 10.0.19042.1466], locale en-US)
[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.2)
[√] Android Studio (version 2020.3)
[√] VS Code (version 1.62.3)
[√] Connected device (5 available)
[√] HTTP Host Availability

• No issues found!

```

</details>

**Comment by chrisnorman7** [comment]
This is still present in version 3.1.0.

Is there any news on or or when this will be looked into? According to [the accessibility page](https://docs.flutter.dev/development/accessibility-and-localization/accessibility):

> We strongly encourage you to include an accessibility checklist as a key criteria before shipping your app. Flutter is
> committed to supporting developers in making their apps more accessible, and includes first-class framework support for
> accessibility in addition to that provided by the underlying operating system...

I believe that - certainly in the base of blind users - app developers cannot currently meet that goal, because Flutter doesn't allow the entering of text in an accessible manner on Windows.

I understand if there's a reason this isn't being worked on, and I get that Flutter is massive, and there's bigger fish to fry, especially with blind users making up such a small minority of users. That said, when blind people pass over Flutter because of this issue, I'd like to be able to give a suitable Flutter defence.

Cheers,

Chris

**Comment by cbracken** [comment]
@chrisnorman7 thanks for checking in on this. 

I've added this (and will be adding a few more Windows accessibility bugs to our burndown list for Q4. Q3 is packed pretty tight right now but I think we've got a collection of enough issues to run an all-out a11y fixit pass and hit as many as we can while they're fresh in our heads in Q4. 

Thanks again for filing these -- I really appreciate the detail that you've been putting into the reports.

**Comment by chrisnorman7** [comment]
Hi,
Thank you for adding them to your list. These bugs - in particular this one
- are the final barriers to "proper" Flutter accessibility. Certainly on
Windows.

I'll see if I can rustle up some Linux ones for you all to work on next
year! :P

Take care,

Chris Norman

**Comment by yaakovschectman** [comment]
I am not observing all this same behavior in the repro on Windows. When typing a new character only the new character is read, and the new text is read if the user does anything to trigger the text field to be re-read (e.g. Narrator+UP or hover cursor)

**Comment by yaakovschectman** [comment]
In https://github.com/flutter/engine/blob/main/shell/platform/windows/accessibility_bridge_delegate_windows.cc#L95 it appears we effectively just ignore `DOCUMENT_SELECTION_CHANGED` events. I see a few possible Windows event constants that may be appropriate, here particularly `EVENT_OBJECT_SELECTION` and `EVENT_OBJECT_TEXTSELECTIONCHANGED`.

**Comment by chrisnorman7** [comment]
@yaakovschectman 
Narrator does that when pressing letter keys in Spotify as well. It's just repeating the key you've pressed, not reading anything meaningful.

Narrator is really only used by people setting up their computers for the first time, before switching to their daily driver. If you wish to test things with a screen reader, I'd recommend [NVDA](https://www.nvaccess.org/), which is free and open source, and used in the wild. Furthermore, you should turn off character echo (insert + 2 with NVDA), since that's where you see the real problems.

Honestly, if someone has such poor site they are using a screen reader, they will probably not be using the mouse to trigger anything, so the arrow keys are far more representative of actual blind users. Obviously I can't speak for people with other disabilities and using other assistive technologies.

**Comment by yaakovschectman** [comment]
So far in investigating this, I have that when a text field is selected/focused, while Inspect and the debug Python console in NVDA report that the text field node is `focused`, the `client (role 0xA)` node that contains it is what NVDA treats as the focused component, so certain edit messages are not being fired.

**Comment by Keithcat1** [comment]
Just tested the code above with the latest Flutter master, NVDA 2023.1. Text field started out unfocused, had to use NVDA's object navigation to get in it, tab didn't work for some reason. It still reads out the entire contents of the text field every time you change it and moving around inside will still not tell you where it's going.

**Comment by aaryan-kumar** [comment]
Hello!
Any update on this?

**Comment by ChrisCRCB** [comment]
Are there any plans to do anything with this? It's been well over 3 years, and we still can't edit text in an accessible way. For a framework which prides itself on "committed to supporting developers in making their apps more accessible", this is pretty poor.

From what I've seen of the framework code, it's not something which can be fixed in pure Dart, or I'd have already done it.

**Comment by martin-robert-fink** [comment]
Still reproduces on 3.47.3 with NVDA 2026.1 on Windows 11 LTSC 24H2. Typing into a `TextField` makes NVDA read the whole content after each keystroke, even with NVDA's typing echo turned off. Left and Right arrows move the caret silently, and Up and Down do not read the line.

Pointers from the engine source at 06a2e2a110, in case they save someone time:

- **Whole-content readout on each keystroke.** Each edit changes the node's value. `AccessibilityBridgeWindows::OnAccessibilityEvent` maps `VALUE_CHANGED` to `ax::mojom::Event::kValueChanged`, which `AXPlatformNodeWin::MojoEventToMSAAEvent` turns into `EVENT_OBJECT_VALUECHANGE`. NVDA's reaction to a value change on the focused edit is to speak the new value.
- **Silent caret moves.** They only update the tree's selection data (`AccessibilityBridge::SetTreeData`). That yields `DOCUMENT_SELECTION_CHANGED`, which is dispatched as `kDocumentSelectionChanged`. That maps to `EVENT_OBJECT_TEXTSELECTIONCHANGED` for MSAA, but to `UIA_Text_TextChangedEventId` rather than `UIA_Text_TextSelectionChangedEventId` in `MojoEventToUIAEvent`. The UIA mapping looks wrong on its face, though I have not confirmed which API NVDA uses for Flutter windows.

As others have noted, there is no workaround from Dart.

**Comment by ChrisCRCB** [comment]
Hi @martin-robert-fink, thank you for weighing in on this.

Honestly, I've got tired of trying to chase this one down and getting no responses beyond confirmations that the problem is there. I really appreciate the digging you did. You got further in your investigation than I did.
